cinema/serializers.py

```diff
@@ -68,13 +78,94 @@ class Meta:
             "movie_title",
             "cinema_hall_name",
             "cinema_hall_capacity",
+            "tickets_available"
         )
 
+    def get_tickets_available(self, obj):
+        total_capacity = obj.cinema_hall.capacity
+        taken_tickets = Ticket.objects.filter(movie_session=obj).count()
+        return total_capacity - taken_tickets
+
 
 class MovieSessionDetailSerializer(MovieSessionSerializer):
     movie = MovieListSerializer(many=False, read_only=True)
     cinema_hall = CinemaHallSerializer(many=False, read_only=True)
+    taken_places = serializers.SerializerMethodField()
 
     class Meta:
         model = MovieSession
-        fields = ("id", "show_time", "movie", "cinema_hall")
+        fields = ("id", "show_time", "movie", "cinema_hall", "taken_places")
+
+    def get_taken_places(self, obj):
+        tickets = Ticket.objects.filter(movie_session=obj)
+        return [{"row": ticket.row, "seat": ticket.seat} for ticket in tickets]
+
+
+class TicketSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Ticket
+        fields = (
+            "id",
+            "row",
+            "seat",
+            "movie_session",
+        )
+
+    def validate(self, attrs):
+        Ticket.validate_ticket(
+            row=attrs["row"],
+            seat=attrs["seat"],
+            movie_session=attrs["movie_session"],
+            error_to_raise=serializers.ValidationError
```

Comment:
The use of `error_to_raise` in the `validate` method is unconventional. Typically, validation methods directly raise `serializers.ValidationError` without passing an error class as a parameter. Consider removing this parameter and directly raising `serializers.ValidationError`.